### PR TITLE
Fix staticcheck on vendor/k8s.io/apiserver/pkg/server/httplog

### PR DESCRIPTION
**Pull request by lauchokyip**
**What type of PR is this?**

/kind cleanup

**What this PR does / why we need it**:
Fix staticheck on `vendor/k8s.io/apiserver/pkg/server/httplog`

Error:
```
vendor/k8s.io/apiserver/pkg/server/httplog/httplog.go:228:15: http.CloseNotifier is deprecated: the CloseNotifier interface predates Go's context package. New code should use Request.Context instead.  (SA1019)

```

**Which issue(s) this PR fixes**:

Part of #92402

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

File: staging/src/k8s.io/apiserver/pkg/server/httplog/httplog.go [hunk 1]
```diff
@@ -224,8 +224,8 @@ func (rl *respLogger) Hijack() (net.Conn, *bufio.ReadWriter, error) {
 }
 
 // CloseNotify implements http.CloseNotifier
-func (rl *respLogger) CloseNotify() <-chan bool {
-	return rl.w.(http.CloseNotifier).CloseNotify()
+func (rl *respLogger) CloseNotify() <-chan struct{} {
```
Comment (caesarxuchao) [inline staging/src/k8s.io/apiserver/pkg/server/httplog/httplog.go:227]: Can we just remove the CloseNotify method?

`respLogger` is private, so probably it's safe to remove.

Comment (lauchokyip) [reply staging/src/k8s.io/apiserver/pkg/server/httplog/httplog.go]: Removed

**Comment by caesarxuchao** [comment]
There are many "request connection cannot be hijacked" error in the pull-kubernetse-e2e-kind. I don't understand why removing the CloseNotify method would lead to this error. @lauchokyip can you double check?

**Comment by liggitt** [comment]
there are places where we check if the wrapped handler implements CloseNotifier/Flusher/Hijacker to determine whether to wrap with a handler that does all of those:

```
		_, cn := w.(http.CloseNotifier)
		_, fl := w.(http.Flusher)
		_, hj := w.(http.Hijacker)
		if cn && fl && hj {
```

dropping this method means this handler no longer implements that interface

**Comment by lauchokyip** [comment]
@liggitt , should I just use `lint: ignore` then

**Comment by lauchokyip** [comment]
or changing it to `Request.Context`

**Comment by liggitt** [comment]
ignore the warning (and add a type assertion to make sure we don't try to drop this again)

**Comment by lauchokyip** [comment]
@liggitt I changed it, can you check if I did it correctly ? Thanks.

File: staging/src/k8s.io/apiserver/pkg/server/httplog/httplog.go [hunk 2]
```diff
@@ -223,8 +223,9 @@ func (rl *respLogger) Hijack() (net.Conn, *bufio.ReadWriter, error) {
 	return rl.w.(http.Hijacker).Hijack()
 }
 
-// CloseNotify implements http.CloseNotifier
+// CloseNotify implements http.CloseNotify.
 func (rl *respLogger) CloseNotify() <-chan bool {
+	//lint:ignore SA1019 dropping this method means this handler no longer implements that interface
 	return rl.w.(http.CloseNotifier).CloseNotify()
 }
```
Comment (caesarxuchao) [inline staging/src/k8s.io/apiserver/pkg/server/httplog/httplog.go:230]: Can you add an interface implementation check to make sure we don't drop this method again? E.g.,
```golang
var _ http.CloseNotifier = &respLogger{}
```

File: staging/src/k8s.io/apiserver/pkg/server/httplog/httplog.go [hunk 3]
```diff
@@ -223,8 +223,9 @@ func (rl *respLogger) Hijack() (net.Conn, *bufio.ReadWriter, error) {
 	return rl.w.(http.Hijacker).Hijack()
 }
 
-// CloseNotify implements http.CloseNotifier
+// CloseNotify implements http.CloseNotify.
 func (rl *respLogger) CloseNotify() <-chan bool {
+	//lint:ignore SA1019 dropping this method means this handler no longer implements that interface
```
Comment (caesarxuchao) [inline staging/src/k8s.io/apiserver/pkg/server/httplog/httplog.go:228]: How about
```golang
//lint:ignore SA1019 There are places in the code base requiring the CloseNotifier interface to be implemented.
```

**Comment by lauchokyip** [comment]
Done @caesarxuchao

**Pull request merged**